src/spoke/interfaces/IVaultSpoke.sol

```diff
@@ -0,0 +1,117 @@
+  function depositWithSig(
+    EIP712Types.VaultDeposit calldata params,
+    bytes calldata signature
+  ) external returns (uint256);
+
+  /// @notice Mints shares of the vault with a signature.
+  /// @dev Uses keyed-nonces where for each key's namespace nonce is consumed sequentially.
+  /// @param params The parameters for the mint.
+  /// @param signature The EIP712-typed signed bytes for the mint.
+  /// @return The amount of assets deposited.
+  function mintWithSig(
+    EIP712Types.VaultMint calldata params,
+    bytes calldata signature
+  ) external returns (uint256);
+
+  /// @notice Withdraws assets from the vault with a signature.
+  /// @dev Uses keyed-nonces where for each key's namespace nonce is consumed sequentially.
+  /// @param params The parameters for the withdraw.
+  /// @param signature The EIP712-typed signed bytes for the withdraw.
+  /// @return The amount of shares burnt.
+  function withdrawWithSig(
+    EIP712Types.VaultWithdraw calldata params,
+    bytes calldata signature
+  ) external returns (uint256);
+
+  /// @notice Redeems shares from the vault with a signature.
+  /// @dev Uses keyed-nonces where for each key's namespace nonce is consumed sequentially.
+  /// @param params The parameters for the redeem.
+  /// @param signature The EIP712-typed signed bytes for the redeem.
+  /// @return The amount of assets burnt.
+  function redeemWithSig(
+    EIP712Types.VaultRedeem calldata params,
+    bytes calldata signature
+  ) external returns (uint256);
+
+  /// @notice Deposits assets into the vault with an underlying asset ERC2612 typed permit.
+  /// @param assets The amount of assets to deposit.
+  /// @param receiver The receiver of the shares.
+  /// @param deadline The deadline of the permit.
+  /// @param v The v value of the permit.
+  /// @param r The r value of the permit.
+  /// @param s The s value of the permit.
+  /// @return The amount of shares minted.
+  function depositWithPermit(
+    uint256 assets,
+    address receiver,
+    uint256 deadline,
+    uint8 v,
+    bytes32 r,
+    bytes32 s
+  ) external returns (uint256);
+
+  /// @notice Resets the allowance of an owner for the caller.
+  /// @param owner The owner of the allowance to renounce.
+  function renounceAllowance(address owner) external;
+
+  /// @notice Returns the address of the associated Hub.
+  function hub() external view returns (address);
+
+  /// @notice Returns the identifier of the associated asset.
+  function assetId() external view returns (uint256);
+
+  /// @notice Returns the maximum allowed spoke cap.
+  function MAX_ALLOWED_SPOKE_CAP() external view returns (uint40);
+
+  /// @notice Returns the nonce key for the share token permit EIP-712 typed signatures.
+  /// @dev Share token permits nonces are always set at this specific key namespace.
+  /// Once the 2 ^ 64 - 1 nonces are used, the nonce at this namespace will overflow and reset to 0; unexpired permits can be replayed then.
```

Comment:
too long, and
feels important to add this to the `permit` function itself, so thinking that we could have our own natspec docs there instead of inheriting